File: Day-148/Smaller or equal elements.py
class Solution:
    # @param A : list of integers
    # @param B : integer
    # @return an integer
    def solve(self, A, B):
        keyCount = A.count(B)
        index = 0
        if keyCount == 0:
            for i in range(len(A)):
                if A[i] < B:
                    continue
                else:
                    index = i - 1
                    break
            else:
                index = len(A) - 1
        else:
            count = 0
            for i in range(len(A)):
                if A[i] == B:
                    count += 1
                if count == keyCount:
                    index = i
                    break
        return len(A[:index+1])

File: Day-150/Smaller or equal elements.js
module.exports = { 
 //param A : array of integers
 //param B : integer
 //return an integer
	solve : function(A, B){
       var len=A.length-1;
       var ans=A.length;
        var st=0,en=len;
        if(A[st]>B) return 0;
        while(st<=en){
            var mid=Math.floor(st+(en-st)/2);
            if(A[mid]<=B) {
                st=mid+1;
                ans=mid+1;
                }
            else en=mid-1;
        }
        return ans;
	}
};
